apps/condo/domains/miniapp/access/B2BAccessToken.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Generated by `createschema miniapp.B2BAccessToken 'sessionId:Text; context:Relationship:B2BAppContext:CASCADE; rightSet:Relationship:B2BAppAccessRightSet:CASCADE; expiresAt:Text;'`
+ */
+
+const { userIsAdmin, isSpecificQuery } = require('@open-condo/keystone/access')
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+
+const { canReadObjectsAsB2BAppServiceUser, canManageObjectsAsB2BAppServiceUser } = require('@condo/domains/miniapp/utils/b2bAppServiceUserAccess')
+const { SERVICE } = require('@condo/domains/user/constants/common')
+
+async function canReadB2BAccessTokens (args) {
```

Comment:
I guess you should protect filtering by any sensitive fields here, like we do in user...

So queries like `allB2BAccessTokens(where: { token_starts_with: })` are banned, because you can brute force them then

Reply:
Virtual fields are not queriable
Admin needs to be able to query by `sessionId` I'm sure
Service users anyway can see only their's tokens, so it is pointless
But if we restrict to them access for `sessionId`, which is surely not needed for them, it might have some cons, so made these 2 restrictions for them